wolfcrypt/src/sha512.c

```diff
@@ -1402,22 +1402,17 @@ static WC_INLINE int Sha512Final(wc_Sha512* sha512)
 
 static int Sha512FinalRaw(wc_Sha512* sha512, byte* hash, size_t digestSz)
 {
-#ifdef LITTLE_ENDIAN_ORDER
-    word64 digest[WC_SHA512_DIGEST_SIZE / sizeof(word64)];
-#endif
-
     if (sha512 == NULL || hash == NULL) {
         return BAD_FUNC_ARG;
     }
 
 #ifdef LITTLE_ENDIAN_ORDER
-    ByteReverseWords64((word64*)digest, (word64*)sha512->digest,
+    ByteReverseWords64((word64*)sha512->digest, (word64*)sha512->digest,
```

Comment:
ByteReverseWords64() doesn't handle unaligned access.

Reply:
The args were indeed 64-bit-aligned (`word64` arrays), which is probably why the tests all passed.  I've removed the casts to make clear that the accesses are safe.